docs/project/cpp_style_guide.md

```diff
@@ -147,6 +147,9 @@ these.
         although these guidelines differ slightly.
 -   Always mark constructors `explicit` unless there's a specific reason to
     support implicit or `{}` initialization.
+-   When storing unowned objects as members, prefer to pass by reference and
+    store as a pointer. For example, `Foo* foo_;` and
+    `Bar(Foo& foo) : foo_(&foo)`.
```

Comment:
Based on further discussion, let's keep the storing part but add some more explicit guidance for passing. This is allowed and *slightly* encouraged in the Google style guide, but I think we can be a bit more direct and explicit for Carbon:

```suggestion
-   When passing a reference to an unowned object as an argument, use a reference
    unless one of the following cases applies:
    -    If it is optional, use a pointer and document that it may be null.
    -    If it is captured and must outlive the call expression itself, use a
         pointer an document that it must not be null (unless it is also optional).
-   When storing a reference to an unowned object as a member, prefer to store as a
    pointer. For example, `Foo* foo_;` and `Bar(Foo* foo) : foo_(foo)`.
```

Reply:
Pulled, and made some edits (including "when passing/storing a reference to an unowned object" -> "when passing/storing an object's address" in order to avoid different meanings of "reference")